code/datums/mapgen/CaveGenerator.dm

```diff
@@ -193,3 +300,67 @@
 	var/message = "[name] terrain population finished in [(REALTIMEOFDAY - start_time)/10]s!"
 	to_chat(world, span_boldannounce("[message]"))
 	log_world(message)
+
+
+/**
+ * This handles the population of terrain with biomes. Should only be called by
+ * `populate_terrain()`, if you find yourself calling this, you're probably not
+ * doing it right.
+ */
+/datum/map_generator/cave_generator/proc/populate_terrain_with_biomes(list/turfs, area/generate_in)
```

Comment:
They're unique procs because they're called only once by the populate_terrain() proc, beforehand they were implemented as a subtype of `/datum/map_generator/cave_generator`, `/biome`, but at oranges' request, I moved them to this so that it's all apart of the cave generator instead.